prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,229 @@
+
+1.) Utility to check sender test compendiums against current mapping table<br>
+2.) Utility to automatically update tables from Value Set Authority Center API (https://www.nlm.nih.gov/vsac/support/usingvsac/vsacfhirapi.html) <br>
+3.) Reporting mechanism to inform RS/sender when unmapped value was found.<br>
+4.) Mechanism to automate which conditions each state considers reportable.
+
+## Design
+
+
+### Condition Mapping Table
+
+The condition mapping table will be made up of CSTE ValueSets and contain the following columns:
+
+| Column Name                   | Description                                   | Example                                                                        |
+|-------------------------------|-----------------------------------------------|--------------------------------------------------------------------------------|
+| Member OID                    | ValueSet Identifier                           | 2.16.840.1.113762.1.4.1146.239                                                 |
+| Name                          | ValueSet Name                                 | Chlamydia trachomatis Infection (Tests for Chlamydia trachomatis Nucleic Acid) |
+| Code                          | LOINC or SNOMED coded value                   | 16601-7                                                                        |
+| Descriptor                    | LOINC or SNOMED term descriptio               | Chlamydia trachomatis rRNA [Presence] in Urine by Probe                        |
+| Code System                   | Indicates whether code is LOINC or SNOMED     | LOINC                                                                          |
+| Version                       | LOINC or SNOMED release version               | 2.74                                                                           |
+| Status                        | Indicates if code is active or depracated     | Active                                                                         |
+| Condition Name                | Name of associated reportable condition       | Chlamydia trachomatis infection (disorder)                                     |
+| Condition Code                | SONMED value associated with condition        | 240589008                                                                      | 
+| Condition Code System         | System used for condition code                | SNOMEDCT                                                                       |
+| Condition Code System Version | SNOMED version associated with condition code | 2023-03                                                                        |
+
+The column names are taken directly from the RCTC spreadsheet.Both LOINC and SNOMED codes are combined in this table and can be identified by column "Code System".
+
+### Supplemental Condition Mapping Table
```

Comment:
Could you expand a bit more on the value of of the supplemental table vs adding custom mappings directly to the Condition Mapping Table?

Reply:
Yes, the idea is that the condition mapping table stays true to the RCTC and CSTE valuesets and we can create tooling to automatically update that table. That gets more complicated if we have to worry about maintaining other values as opposed to just updating the entire table, but I can also see the argument that says if we are creating something custom to update the table anyways why not just build in the functionality to keep any values we added.

Reply:
Ultimately the hope is that as CSTE and RCKMS expand more into ELR space (which is on their roadmap) we have a common "source of truth"

Reply:
yeah I definitely think it's possible to add all to the same table. We'd just need a way to identify the source of the record(manual upload vs external import) which could just be a column. The only reason I can see to create a supplemental table is if additional/different columns are needed. I could see columns for created date, uploaded by, and maybe expiry date. I think most of those would be covered by the lookup table version for the imported table but could be useful for the manual upload records.

Reply:
We also need to ensure we are tracking changes to the table, right? I don't see anything in the doc for that?

Reply:
Yes, the tables are versioned and we could also add a column for which version of the valueset we are referencing, however I'm not sure what value this gets us?.

Reply:
Also @brick-green I went back and forth on having things be in one table or two (or even more). I don't think there's a right or wrong answer, just whatever most people think is easier to work with is what I would vote for.

Reply:
@GilmoreA6 maybe this could be a eng sync topic?

Reply:
Not a bad idea. Maybe see what people's thoughts are on table maintenance in general.

Reply:
@GilmoreA6 Was this discussed at an eng sync? I missed one or two so wasn't sure